File: src/main/java/failchat/funstream/AnnounceMessageFilter.java
package failchat.funstream;

import failchat.core.MessageFilter;

import java.util.logging.Logger;

public class AnnounceMessageFilter implements MessageFilter<FsMessage> {
    private static final Logger logger = Logger.getLogger(AnnounceMessageFilter.class.getName());

    @Override
    public boolean filterMessage(FsMessage message) {
        if (message.getType().equals("announce")) {
            logger.fine("Announce message filtered: " + message.getText());
            return true;
        } else {
            return false;
        }
    }
}

File: src/main/kotlin/failchat/peka2tv/AnnounceMessageFilter.kt
package failchat.peka2tv

import failchat.chat.MessageFilter
import mu.KotlinLogging

class AnnounceMessageFilter : MessageFilter<Peka2tvMessage> {

    private companion object {
        val logger = KotlinLogging.logger {}
    }

    override fun filterMessage(message: Peka2tvMessage): Boolean {
        if (message.type == "announce") {
            logger.debug("Announce message filtered: {}", message.text)
            return true
        } else {
            return false
        }
    }

}
